### ISSUE TITLE
bat panics on Ruby files with unindented heredocs

### ISSUE THREAD

**Issue by sharkdp** [sharkdp/bat#914]
When calling `bat` on this Ruby file
``` ruby
<<END_DESC
Test
END_DESC
```
it panics with:
```
thread 'main' panicked at
  'regex string should be pre-tested: Error(-208, invalid backref number/name)',
  …/syntect-4.1.0/src/parsing/regex.rs:70:17
```

found with the Python script in #913.

**Comment by sharkdp** [comment]
This is caused by the block here: https://github.com/sublimehq/Packages/blob/f36b8f807d5f30d2b8ef639232a9fc5960f550fa/Ruby/Ruby.sublime-syntax#L1914-L1917
``` yaml
  unindented-heredoc-end:
    - match: ^\1$ # HEREDOC delimiter
      scope: string.unquoted.heredoc.ruby punctuation.definition.string.end.ruby
      pop: true
```

This block seems to reference the heredoc name from the opening part
``` yaml
  heredoc-ruby:
    - match: <<[-~]["`]?({{heredoc_type_ruby}})["`]?
      scope: string.unquoted.heredoc.ruby punctuation.definition.string.begin.ruby
      push: [heredoc-ruby-indented-interpolated, trailing-heredoc-start]
    - match: <<[-~]'({{heredoc_type_ruby}})'
      scope: string.unquoted.heredoc.ruby punctuation.definition.string.begin.ruby
      push: [heredoc-ruby-indented-literal, trailing-heredoc-start]
    - match: <<["`]?({{heredoc_type_ruby}})["`]?
      scope: string.unquoted.heredoc.ruby punctuation.definition.string.begin.ruby
      push: [heredoc-ruby-unindented-interpolated, trailing-heredoc-start]
    - match: <<'({{heredoc_type_ruby}})'
      scope: string.unquoted.heredoc.ruby punctuation.definition.string.begin.ruby
      push: [heredoc-ruby-unindented-literal, trailing-heredoc-start]
```

but for some reason, the `\1` reference can not be resolved.

**Comment by keith-hall** [comment]
this looks like an instance of https://github.com/trishume/syntect/issues/104 to me

**Comment by sharkdp** [comment]
Certainly seems like it - thank you for the reference!

**Comment by sharkdp** [comment]
closed via #929
